### PR TITLE
feat: mssql connection

### PR DESCRIPTION
**Pull request by AyushAgrawal-A2**

### CODE REVIEW

File: quadratic-rust-shared/src/sql/mssql_connection.rs [hunk 1]
```diff
@@ -0,0 +1,613 @@
...
+            .await
+            .map_err(|e| SharedError::Sql(Sql::Schema(e.to_string())))?;
+
+        let mut schema = DatabaseSchema {
+            database: self.database.to_owned().unwrap_or_default(),
+            tables: BTreeMap::new(),
+        };
+
+        for row in rows {
+            let schema_name: &str = row.get(1).unwrap_or_default();
```
Comment (ddimaria) [inline quadratic-rust-shared/src/sql/mssql_connection.rs:193]: Default might be an empty string here.  Maybe some other placeholder would be appropriate (e.g. `Unknown Schema - #num`, `Unknown Column - #num`), where `#num` is the loop index.

File: quadratic-rust-shared/src/sql/mssql_connection.rs [hunk 2]
```diff
@@ -0,0 +1,613 @@
...
+                }
+            },
+        )
+    }
+}
+
+#[macro_export]
+macro_rules! convert_mssql_type {
+    ( $kind:ty, $column_data:ident ) => {{
+        <$kind>::from_sql($column_data)
```
Comment (ddimaria) [inline quadratic-rust-shared/src/sql/mssql_connection.rs:273]: Same notes here on the default value.

File: quadratic-rust-shared/src/sql/mod.rs [hunk 3]
```diff
@@ -270,14 +88,12 @@ pub trait Connection {
...
-            .columns()
-            .iter()
+        let fields = Self::row_columns(&data[0])
             .enumerate()
             .map(|(index, col)| {
                 Field::new(
-                    col.name().to_string(),
-                    cols[index].data_type().to_owned(),
+                    Self::column_name(&col).to_string(),
+                    cols[index].data_type().clone(),
```
Comment (ddimaria) [inline quadratic-rust-shared/src/sql/mod.rs:96]: Nit: I'll write clones when starting a PR, but then remove them or convert to `to_owned()` to signal the intentional copy.  It's not that important here, but that's the pattern I've been doing.

**Comment by davidkircos** [comment]
set default username to `admin`
CleanShot 2024-08-26 at 21 26 49

**Comment by davidkircos** [comment]
Should a database name be required? I believe by default AWS RDS does not create a database

CleanShot 2024-08-26 at 21 27 50

**Comment by davidkircos** [comment]
Schema viewer does not load 

CleanShot 2024-08-26 at 21 37 11

**Comment by davidkircos** [comment]
SQL Syntax highlighting doesn't work for MSSQL 
CleanShot 2024-08-26 at 21 39 19

File: quadratic-rust-shared/src/sql/mssql_connection.rs (hunk 2 above)
Comment (AyushAgrawal-A2) [reply quadratic-rust-shared/src/sql/mssql_connection.rs:273]: switched default value to ArrowType::Void

File: quadratic-rust-shared/src/sql/mssql_connection.rs (hunk 1 above)
Comment (AyushAgrawal-A2) [reply quadratic-rust-shared/src/sql/mssql_connection.rs:193]: done

**Comment by AyushAgrawal-A2** [comment]
>Should a database name be required? I believe by default AWS RDS does not create a database

it defaults to `master` if not provided, i have made it optional now

**Review by ddimaria — APPROVED** [review]

**Pull request merged**